### DISCUSSION TITLE
Simplify downloader?

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#10625] [Ideas]
In the context of the integrated downloader two ideas came across my mind, and I would like to gather some feedback for them:

- Automatically download theme files for maps (if none installed)

  Some maps work better with a specific theme installed. For OpenAndroMaps this is `Elevate`, for Freizeitkarte one of the `Freizeitkarte` themes, and similar for Hylly. Currently the downloader checks if one of the supported/required themes is already installed, and if not, asks the user whether they want to install it as well.

  Proposal: Don't ask, just check, and auto-install, if none found.
  (Theme files are pretty small, typically only 300k - 500k, so no relevant extra load, compared to the size of a map file.)

- Automatically overwrite existing files

  Currently the downloader does a "does the file already exist" check after download completion and asks the user how to proceed, if the file already exists. User can select cancel / overwrite / save under a different name.

  Proposal: Don't ask, simply overwrite existing file after successful download.
  (Actually, I can't think of a reason why a user does not want to overwrite an existing file, except the rare case where a map from two different source have the same name. OAM maps get "(OAM)" suffix, which reduces possible conflicts.)

Reasoning behind those proposals:
- I would like to reduce intermediate steps / user interaction, which is not needed in most cases.
- If we want to put download and transfer completely into the background, as @fm-sys  already has started to work on (see #10310), there should be no further user interaction after the initial check.

What do you think?

**Comment by fm-sys** [comment]
This would indeed simplify the migration to a background service a lot. 

https://github.com/cgeo/cgeo/pull/10310 is even fully functional (as long as non of these special cases occur). So if someone (probably @moving-bits, as I'm not really firm with the code) would rewrite the downloader that definitely no user action is needed after starting the download, I think that I could get the background service working quite quickly...

**Reply by moving-bits** [reply]
I would be willing to adapt the downloader accordingly, but let's wait a bit if there are any objections.

**Comment by moving-bits** [comment]
As there seem to be no objections I'll create issues for it and start working on it.
